src/pages/patientView/mutation/PatientViewMutationTable.tsx

```diff
@@ -427,15 +449,30 @@ export default class PatientViewMutationTable extends React.Component<PatientVie
     }
 
     @computed private get columns():Column<Mutation[]>[] {
-        return this.props.columns.map((columnEnum:MutationTableColumn)=>this._columns[columnEnum]);
+        return this.props.columns.reduce((columns:Column<Mutation[]>[], next:MutationTableColumn)=>{
+            let shouldAdd = true;
+            if (next === MutationTableColumn.MRNA_EXPR &&
+                (!this.props.mrnaExprRankGeneticProfileId
+                || this.getSamples().length > 1)) {
+                shouldAdd = false;
+            } else if (next === MutationTableColumn.TUMORS && this.getSamples().length < 2) {
+                shouldAdd = false;
+            } else if (next === MutationTableColumn.COPY_NUM && !this.props.discreteCNAGeneticProfileId) {
+                shouldAdd = false;
+            }
+            if (shouldAdd) {
```

Comment:
Hmm, looks like the PR has changed, we don't have the `store` here anymore. Defining generic parameters for `shouldAdd` function is not trivial in this case :)